### ISSUE TITLE
Problem updating lucene index database

### ISSUE THREAD

**Issue by fairmonk** [oracle/opengrok#1301]
Hi, I'm running opengrok-0.13-rc5
when reindexing, I get this error at some point:
01:14:55 Dec 19, 2016 1:14:42 AM org.opensolaris.opengrok.index.IndexDatabase optimize
01:14:55 INFO: Optimizing the index ... 
01:14:55 Dec 19, 2016 1:14:42 AM org.opensolaris.opengrok.index.IndexDatabase$2 run
**01:14:55 SEVERE: Problem updating lucene index database:** 
**01:14:55 java.lang.IllegalArgumentException: Directory MMapDirectory@C:\opengrok\Data\index\PC-trunk lockFactory=org.apache.lucene.store.NoLockFactory@319220f6 still has pending deleted files; cannot initialize IndexWriter**
01:14:55 	at org.apache.lucene.index.IndexWriter.<init>(IndexWriter.java:782)
01:14:55 	at org.opensolaris.opengrok.index.IndexDatabase.optimize(IndexDatabase.java:511)
01:14:55 	at org.opensolaris.opengrok.index.IndexDatabase.update(IndexDatabase.java:439)
01:14:55 	at org.opensolaris.opengrok.index.IndexDatabase$2.run(IndexDatabase.java:184)
01:14:55 	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
01:14:55 	at java.util.concurrent.FutureTask.run(Unknown Source)
01:14:55 	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
01:14:55 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
01:14:55 	at java.lang.Thread.run(Unknown Source)
01:14:55 
01:14:56 Dec 19, 2016 1:14:43 AM org.opensolaris.opengrok.index.DefaultIndexChangedListener fileRemove

I suppose the index failed to update the Delta, Is there something I can do to mitigate it?

**Comment by tarzanek** [comment]
can you write down exact reproduction steps please?
(like step by step)
We should flush the directory upon closing, so it looks like the previous index failed for some unknown reason (which I'd like to detect from your reproduction steps)

**Comment by fairmonk** [comment]
- So I checkout from SVN something like 17 projects (via Jenkins SVN plugin)
- then I run this command to index/re-index (via Jenkins windows batch plugin):
java -Xmx28g -jar C:\opengrok\opengrok-0.13-rc5\lib\opengrok.jar -W C:\opengrok\Data\configuration.xml -c C:\opengrok\ctags58\ctags.exe -P -S -v -s C:\opengrok\src -d C:\opengrok\Data -i *.zip -i *.tmp -i *.db -i *.jar -i *.class -i *.dylib -i *.pdb -i *.har -i *.tar.gz -i *.bin -i *.pdf -i *.doc -i *.msp -i *.msi -i f:regress-274888.js -i f:test553 -i f:test56 -i d:openjdk8-b132 -i d:.svn -G -L polished -a on -T 8
-  a restart to Tomcat at the end
That's it (job runs every 24h)

One thing I noticed: last two executions produced this SEVERE error, I just ran the job again and  error didn't reproduce. So I'm not sure what's the missing piece in the puzzle here to reproduce the error.
If you think there is more to pitch in here, I can provide additional information.
